### PR TITLE
WT-11416 Create unit test for txn_mod_compare

### PR DESCRIPTION
**Pull request by clarissecheah**

### CODE REVIEW

File: test/unittest/tests/test_prepare_mod_sort.cpp [hunk 1]
```diff
@@ -0,0 +1,370 @@
...
+TEST_CASE("Different column store keys test", "[mod_compare]")
+{
+    WT_BTREE btrees[12];
+    WT_TXN_OP ops[8];
+    int ret = 0;
+
+    for (int i = 0; i < 6; i++)
+        init_btree(&btrees[i], BTREE_COL_VAR, i);
+
+    // Randomly pick btrees and assign random recnos to the ops.
```
Comment (luke-pearson) [inline test/unittest/tests/test_prepare_mod_sort.cpp:360]: pick -> choose

File: test/unittest/tests/test_prepare_mod_sort.cpp [hunk 2]
```diff
@@ -0,0 +1,370 @@
...
+{
+    op->btree = btree;
+    op->type = type;
+    if (op->type == WT_TXN_OP_BASIC_COL || op->type == WT_TXN_OP_INMEM_COL) {
+        REQUIRE(recno != WT_RECNO_OOB);
+        op->u.op_col.recno = recno;
+    } else if (op->type == WT_TXN_OP_BASIC_ROW || op->type == WT_TXN_OP_INMEM_ROW) {
+        REQUIRE(key != NULL);
+        op->u.op_row.key = *key;
+    } else {
```
Comment (luke-pearson) [inline test/unittest/tests/test_prepare_mod_sort.cpp:108]: Remove brackets from final case.

File: test/unittest/tests/test_prepare_mod_sort.cpp [hunk 3]
```diff
@@ -0,0 +1,370 @@
...
+
+    allocate_key_space(session, key_sz, keys);
+
+    for (int i = 0; i < 6; i++)
+        init_key(keys[i], random_keys(3));
+
+    init_btree(&btrees[0], BTREE_COL_VAR, 1);
+    init_btree(&btrees[1], BTREE_ROW, 2);
+
+    // Column operations
```
Comment (luke-pearson) [inline test/unittest/tests/test_prepare_mod_sort.cpp:219]: Comments need fullstops.

File: test/unittest/tests/test_prepare_mod_sort.cpp (hunk 2 above)
Comment (clarissecheah) [reply test/unittest/tests/test_prepare_mod_sort.cpp]: [b1a4e7f](https://github.com/wiredtiger/wiredtiger/pull/9522/commits/b1a4e7fd4c59d380f5ae6e9dceae4ff904497b43)

File: test/unittest/tests/test_prepare_mod_sort.cpp (hunk 1 above)
Comment (clarissecheah) [reply test/unittest/tests/test_prepare_mod_sort.cpp]: [b1a4e7f](https://github.com/wiredtiger/wiredtiger/pull/9522/commits/b1a4e7fd4c59d380f5ae6e9dceae4ff904497b43)

File: test/unittest/tests/test_prepare_mod_sort.cpp [hunk 4]
```diff
@@ -0,0 +1,373 @@
...
+    int key_sz = 1;
+    bool ret;
+    WT_ITEM *keys[key_sz];
+
+    allocate_key_space(session, key_sz, keys);
+
+    init_key(keys[0], "1");
+
+    for (int i = 0; i < 6; i++)
+        init_btree(&btrees[i], BTREE_ROW, rand() % 100);
```
Comment (ajmorton) [inline test/unittest/tests/test_prepare_mod_sort.cpp:295]: Is there any issue with picking the same btree id for two btrees? I expect this is ok

File: test/unittest/tests/test_prepare_mod_sort.cpp [hunk 5]
```diff
@@ -0,0 +1,373 @@
...
+
+    for (int i = 0; i < 6; i++)
+        init_btree(&btrees[i], BTREE_ROW, 1);
+    for (int i = 6; i < 12; i++)
+        init_btree(&btrees[i], BTREE_ROW, 2);
+
+    // Operations will have randomly chosen btrees and randomly generated keys.
+    for (int i = 0; i < 12; i++)
+        init_op(
+          &ops[i], &btrees[rand() % 1 + 1], WT_TXN_OP_BASIC_ROW, WT_RECNO_OOB, keys[rand() % 12]);
```
Comment (ajmorton) [inline test/unittest/tests/test_prepare_mod_sort.cpp:342]: Should `&btrees[rand() % 1 + 1]` be `&btrees[rand() % 12]`?

File: test/unittest/tests/test_prepare_mod_sort.cpp [hunk 6]
```diff
@@ -0,0 +1,373 @@
...
+    for (int i = 0; i <= key_sz; i++)
+        __wt_scr_free(session, &keys[i]);
+
+    REQUIRE(ret == true);
+}
+
+// Test sorting on column store keys.
+TEST_CASE("Different column store keys test", "[mod_compare]")
+{
+    WT_BTREE btrees[12];
```
Comment (ajmorton) [inline test/unittest/tests/test_prepare_mod_sort.cpp:356]: This can be `WT_BTREE btrees[6];`

**Review by ajmorton — COMMENTED** [review]
Couple of minor suggestions. Feel to take or leave them as you see fit.

The one comment I'd like to draw attention to is the `rand() % 1` case

File: test/unittest/tests/test_prepare_mod_sort.cpp (hunk 5 above)
Comment (clarissecheah) [reply test/unittest/tests/test_prepare_mod_sort.cpp:342]: We only have btree id's 1 and 2 available here, which is why I used `&btrees[rand() % 1 + 1`

File: test/unittest/tests/test_prepare_mod_sort.cpp (hunk 4 above)
Comment (clarissecheah) [reply test/unittest/tests/test_prepare_mod_sort.cpp:295]: You're right, that should be fine.

File: test/unittest/tests/test_prepare_mod_sort.cpp [hunk 7]
```diff
@@ -0,0 +1,375 @@
...
+
+// Test sorting by keyedness, key'd operations all have the same key and recno.
+TEST_CASE("Keyedness sort test", "[mod_compare]")
+{
+    ConnectionWrapper conn(DB_HOME);
+    WT_SESSION_IMPL *session = conn.createSession();
+
+    WT_BTREE btrees[12];
+    WT_TXN_OP ops[12];
+    const int key_sz = 1;
```
Comment (luke-pearson) [inline test/unittest/tests/test_prepare_mod_sort.cpp:288]: A better name for this var would be `key_count`, `key_sz` makes it seem like that's the length of the key but its actually the number of keys.

Comment (clarissecheah) [reply test/unittest/tests/test_prepare_mod_sort.cpp]: Thanks for pointing that out! [30a56d3](https://github.com/wiredtiger/wiredtiger/pull/9522/commits/30a56d3d9e21b18b31ebe673ddbfcb8deee9ccd2)

**Review by ajmorton — APPROVED** [review]
LGTM 👍

**Review by luke-pearson — APPROVED** [review]
LGTM.

**Pull request merged**